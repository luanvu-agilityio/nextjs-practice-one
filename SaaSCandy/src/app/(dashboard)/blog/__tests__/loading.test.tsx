import { render } from '@testing-library/react';
import BlogListLoading from '@/app/(dashboard)/blog/loading';

describe('BlogListLoading - Snapshot Tests', () => {
  it('should match snapshot', () => {
    const { container } = render(<BlogListLoading />);
    expect(container).toMatchSnapshot();
  });

  it('should render all skeleton elements', () => {
    const { container } = render(<BlogListLoading />);
    const skeletons = container.querySelectorAll(
      '.animate-pulse, [class*="Skeleton"]'
    );

    // Verify multiple skeleton elements are present
    expect(skeletons.length).toBeGreaterThan(0);
  });
});
