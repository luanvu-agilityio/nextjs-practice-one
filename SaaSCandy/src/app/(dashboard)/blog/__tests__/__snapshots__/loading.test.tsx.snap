// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BlogListLoading - Snapshot Tests should match snapshot 1`] = `
<div>
  <section
    class="py-12 sm:py-16 bg-white"
  >
    <div
      class="max-w-[1296px] mx-auto px-4 sm:px-6 lg:px-0 "
    >
      <div
        class="grid sm:grid-cols-2 lg:grid-cols-3 gap-6"
      >
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
        <article
          class="group animate-pulse"
        >
          <div
            class="block rounded-lg overflow-hidden bg-white"
          >
            <div
              class="w-full h-[284px] bg-gray-200"
            />
            <div
              class="flex flex-col gap-4 py-10"
            >
              <div
                class="space-y-2"
              >
                <div
                  class="h-6 bg-gray-200 rounded w-full"
                />
                <div
                  class="h-6 bg-gray-200 rounded w-3/4"
                />
              </div>
              <div
                class="h-3 bg-gray-200 rounded w-20"
              />
            </div>
          </div>
        </article>
      </div>
    </div>
  </section>
</div>
`;
